### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by JoshLmao**
Login system implemented with the current developer database

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Looks all good, can't see any issues in the code. Really like the work with the encryption too.

File: api/user/login/index.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+$conn = mysqli_connect($SERVER_LOCATION, $SERVER_USERNAME, $SERVER_PASSWORD, $DB_NAME);
+if (!$conn) {
+    die("Unable to open connection - " . mysqli_connect_error());
+}
+// echo "Successfully connected <br/>";
+
+$email = $postData['email'];
+$sql = "SELECT * FROM $USER_TABLE WHERE admin_email='$email'";
```
Comment (Philwlv) [inline api/user/login/index.php]: `$sql` variable vulnerable to injection attack with `$email` inserted directly into the query. I would advise the following,

Validate email & Parameterise query:
`if  (!empty($email= filter_var($email, FILTER_VALIDATE_EMAIL)){`
`    $sql = ("SELECT * FROM $USER_TABLE WHERE admin_email= ?");`
`    $sql->bind_param("s", $email);`
`    $result = $conn->query($sql);`
`    }`

File: api/user/login/index.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        $user->admin_email = $row["admin_email"];
+        $user->admin_type = $row["admin_type"];
+        $user->admin_password = $row["admin_password"];
+
+        // Add asset to array
+        $db_array[] = $user;
+    }
+
+    $password = $postData['password'];
+    if ($email == $user->admin_email && $password == $user->admin_password)
```
Comment (Philwlv) [inline api/user/login/index.php]: When comparing password, no passwords should be used. A comparison of the hash of the passwords should be compared. PHP uses functions `password_hash()` and `password_verify()` see [php.net](https://www.php.net/manual/en/book.password.php)

File: api/user/password-change/index.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+$conn = mysqli_connect($SERVER_LOCATION, $SERVER_USERNAME, $SERVER_PASSWORD, $DB_NAME);
+if (!$conn) {
+    die("Unable to open connection - " . mysqli_connect_error());
+}
+
+$user_id = $post_data["id"];
+/// SQL Query command to select all from the 'assets' table
+$sql = "SELECT * FROM `user` WHERE `admin_id` = " . $user_id;
+$result = $conn->query($sql);
```
Comment (Philwlv) [inline api/user/password-change/index.php]: Query vulnerable to sql injection as `$user_id` is added to the end of the statement.
A Paramatarised statement will inform the database of the separation of the query and the input. Advised code:
`$sql = ("SELECT * FROM 'user' WHERE 'admin_id' = ?");`
`$sql->bind_param("s", $user_id);`
`$result = $conn->query($sql);`

File: api/user/password-change/index.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+if ($result && $result->num_rows > 0) 
+{
+    $row = $result->fetch_assoc();
+    // Check current password matches one in database 
+    if ( $row["admin_password"] == $post_data["pass_current"]) 
+    {
+        // Set new password in db
+        $updateSql = "UPDATE `user` SET `admin_password` = \"" . $post_data["pass_new"] . "\" WHERE `user`.`admin_id` = 1";
+        $result = $conn->query($updateSql);
```
Comment (Philwlv) [inline api/user/password-change/index.php:48]: `$updateSql` Query vulnerable to sql injection as `$post_data["pass_new"]`  is added to the middle of the statement.

**Review by Philwlv — APPROVED** [review]
Love it, I have commented on some sql querys that will need adjusting and on one point where the passwords my need hashing.

**Comment by JoshLmao** [comment]
Thanks for those Phil. I've made those improvements you suggested, you absolute ⭐

**Pull request merged**